src/randoop/ExceptionCheck.java

```diff
@@ -0,0 +1,145 @@
+package randoop;
+
+import java.io.ObjectStreamException;
+import java.lang.reflect.Modifier;
+import java.util.Objects;
+
+/**
+ * An {@code ExceptionCheck} is used to indicate that an exception is expected
+ * at a particular statement in a sequence. Depending on command line arguments
+ * to Randoop, an instance may be either a {@link ExpectedExceptionCheck} or 
+ * {@link EmptyExceptionCheck}. Both will catch the exception if it occurs, but
+ * differ on whether the expectation of the exception is enforced.
+ */
+public abstract class ExceptionCheck implements Check {
+  
+ /**
+   * 
+   */
+  private static final long serialVersionUID = 4806179088639914364L;
+
+  protected final Throwable exception;
+  
+  // Indicates which statement results in the given exception. 
+  protected final int statementIndex;
+
+  public ExceptionCheck(Throwable exception, int statementIndex) {
+    this.exception = exception;
+    this.statementIndex = statementIndex;
+  }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
I just meant there is no state in the implementing classes, they just
provide alternative behaviors based on type.

On Fri, Dec 18, 2015 at 9:23 AM, Michael Ernst <notifications@github.com>
wrote:

> In src/randoop/ExceptionCheck.java
> <https://github.com/randoop/randoop/pull/73#discussion_r48048156>:
>
> > +   *
> > +   */
> > +  private static final long serialVersionUID = 4806179088639914364L;
> > +
> > +  protected final Throwable exception;
> > +
> > +  // Indicates which statement results in the given exception.
> > +  protected final int statementIndex;
> > +
> > +  public ExceptionCheck(Throwable exception, int statementIndex) {
> > +    this.exception = exception;
> > +    this.statementIndex = statementIndex;
> > +  }
> > +
> > +  @Override
> > +  public boolean equals(Object o) {
>
> Sorry, I don't understand your comment.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/randoop/randoop/pull/73/files#r48048156>.
>



-- 
Ben Keller
bjkeller@uw.edu